components/backstage/backstage_controller.tsx

```diff
@@ -0,0 +1,221 @@
+
+import React, {ComponentType, useRef} from 'react';
+import {match, Route, Switch} from 'react-router-dom';
+
+import {UserProfile} from '@mattermost/types/users.js';
+
+import {Team} from '@mattermost/types/teams.js';
+
+import Bots from 'components/integrations/bots';
+import AddBot from 'components/integrations/bots/add_bot';
+import AnnouncementBar from 'components/announcement_bar';
+import SystemNotice from 'components/system_notice';
+import Integrations from 'components/integrations';
+import Emoji from 'components/emoji';
+import AddEmoji from 'components/emoji/add_emoji';
+import InstalledIncomingWebhooks from 'components/integrations/installed_incoming_webhooks';
+import AddIncomingWehook from 'components/integrations/add_incoming_webhook';
+import EditIncomingWebhook from 'components/integrations/edit_incoming_webhook';
+import InstalledOutgoingWebhooks from 'components/integrations/installed_outgoing_webhooks';
+import AddOutgoingWebhook from 'components/integrations/add_outgoing_webhook';
+import EditOutgoingWebhook from 'components/integrations/edit_outgoing_webhook';
+import InstalledOauthApps from 'components/integrations/installed_oauth_apps';
+import AddOauthApp from 'components/integrations/add_oauth_app';
+import EditOauthApp from 'components/integrations/edit_oauth_app';
+import CommandsContainer from 'components/integrations/commands_container';
+import ConfirmIntegration from 'components/integrations/confirm_integration';
+
+import Pluggable from 'plugins/pluggable';
+
+import BackstageSidebar from './components/backstage_sidebar';
+import BackstageNavbar from './components/backstage_navbar';
+
+type ExtraProps = Pick<Props, 'user' | 'team'> & {scrollToTop: () => void}
+
+type BackstageRouteProps = {
+    component: ComponentType<any>;
+    extraProps: ExtraProps;
+    path: string;
+    exact?: boolean;
+}
+
+const BackstageRoute = ({component: Component, extraProps, ...rest}: BackstageRouteProps) => (
+    <Route
+        {...rest}
+        render={(props) => (
+            <Component
+                {...extraProps}
+                {...props}
+            />
+        )}
+    />
+);
+
+type Props = {
+
+    /**
+     * Current user.
+     */
+    user: UserProfile;
+
+    /**
+     * Current team.
+     */
+    team: Team;
+
+    /**
+     * Object from react-router
+     */
+    match: match<{url: string}>;
+
+    siteName?: string;
+    enableCustomEmoji: boolean;
+    enableIncomingWebhooks: boolean;
+    enableOutgoingWebhooks: boolean;
+    enableCommands: boolean;
+    enableOAuthServiceProvider: boolean;
+    canCreateOrDeleteCustomEmoji: boolean;
+    canManageIntegrations: boolean;
+}
+
+const BackstageController = (props: Props) => {
+    const listRef = useRef<HTMLDivElement>(null);
+
+    const scrollToTop = () => {
+        if (listRef.current) {
+            listRef.current.scrollTop = 0;
+        }
+    };
+
+    if (props.team == null || props.user == null) {
+        return <div/>;
+    }
```

Comment:
Slight changes here since I think this is just written this way since it's just old code. There's nothing particularly wrong, but it's an old style

```suggestion
    if (!props.team || !props.user) {
        return null;
    }
```